packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GMSMarker+Userdata.m

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "GMSMarker+Userdata.h"
+
+@implementation GMSMarker (Userdata)
+
+- (void)setMarkerId:(NSString *)markerId {
+  self.userData = @[ markerId ];
+}
+
+- (void)setMarkerID:(NSString *)markerId andClusterManagerId:(NSString *)clusterManagerId {
+  self.userData = @[ markerId, clusterManagerId ];
+}
+
+- (NSString *)getMarkerId {
+  if ([self.userData count] == 0) {
+    return nil;
+  }
+  return self.userData[0];
+}
+
+- (NSString *)getClusterManagerId {
+  if ([self.userData count] != 2) {
+    return nil;
+  }
+
+  NSString *clusterManagerId = self.userData[1];
+  if (clusterManagerId == (id)[NSNull null]) {
```

Comment:
The cast here is working around an abuse of the type system on the previous line'; assigning something that may be `NSNull` to an `NSString*` is incorrect. The fix for the compiler message you got is to not do that, rather than to suppress it with casts. If you don't know the type when you extract to a local variable, the correct type for that variable is `id`.